utilities/src/main/java/ru/study21/jcsv/xxl/io/RandomAccessCachedNioReader.java

```diff
@@ -0,0 +1,88 @@
+package ru.study21.jcsv.xxl.io;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.Random;
+
+// assumes underlying file is not updated
+public class RandomAccessCachedNioReader {
+
+    private final ByteBuffer[] caches;
+    private final long[] positions;
+    private final FileChannel channel;
+    private final CachePolicy cachePolicy;
+
+    public interface CachePolicy {
+        int readCache(long pos) throws IOException; // return index of new cache
+
+        default void didRead(int cacheIndex) { /* do nothing */ } // trigger to calc stats (for LRU / ...)
+
+        enum Type {RR} // TODO: LRU requires heap with decrease-key, use already implemented PairingHeap
```

Comment:
LRU is easy to implement as double-linked list with dedicated head element.
Each element has links to next and previous ones.
And upon access each element is removed from current place and attached to head.
Thus previous of head (aka tail) is one to be deleted.